app/src/main/java/org/simple/clinic/navigation/v2/fragments/BaseScreen.kt

```diff
@@ -19,12 +19,12 @@ import com.spotify.mobius.functions.Consumer
 import com.spotify.mobius.rx2.RxMobius
 import io.reactivex.Observable
 import io.reactivex.ObservableTransformer
-import org.simple.clinic.util.unsafeLazy
 import org.simple.clinic.mobius.eventSources
 import org.simple.clinic.mobius.first
 import org.simple.clinic.navigation.v2.ScreenKey
+import org.simple.clinic.util.unsafeLazy
 
-abstract class BaseScreen<K: ScreenKey, M: Parcelable, E, F>: Fragment() {
+abstract class BaseScreen<K : ScreenKey, B, M : Parcelable, E, F> : Fragment() {
```

Comment:
We can make the binding generic extending `ViewBinding`.

Reply:
Can we do that? I didn't know that the binding classes extend a common type. If they do, that might make some of the stuff simpler. I'll take a look.

Reply:
Yep. unfortunately, it doesn't have `bind` or `inflate` methods but it provides us with `getRoot` method and a way to make sure they only provide binding classes.